src/calculus/AdjointOperator.jl

```diff
@@ -35,6 +35,8 @@ size(L::AdjointOperator) = size(L.A,2), size(L.A,1)
 
 domainType(L::AdjointOperator) = codomainType(L.A)
 codomainType(L::AdjointOperator) = domainType(L.A)
+domainStorageType(L::AdjointOperator) = codomainStorageType(L.A)
+codomainStorageType(L::AdjointOperator) = domainStorageType(L.A)
```

Comment:
Is there any situation where `(co)domainType != (co)domainStorageType`?

Reply:
I'm not sure if I understand your question correctly. If you mean `(co)domainType != eltype((co)domainStorageType)`, then you're right: there should not be any cases when the `(co)domainType` and the `eltype` of `(co)domainStorageType` differ.

Reply:
yes was wondering if we can use only `domainType` and avoid introducing `domainStorageType`

Reply:
was just wondering if that could be simplified, anyway looks good! feel free to merge, thanks!